demos/general/demo_pipeline_voltage_imaging.py

```diff
@@ -105,55 +105,75 @@ def main():
 
 # %% MEMORY MAPPING
-    border_to_0 = 0 if mc.border_nan == 'copy' else mc.border_to_0
-    # you can include the boundaries of the FOV if you used the 'copy' option
-    # during motion correction, although be careful about the components near
-    # the boundaries
+    do_memory_mapping = True
+    if do_memory_mapping:
+        border_to_0 = 0 if mc.border_nan == 'copy' else mc.border_to_0
+        # you can include the boundaries of the FOV if you used the 'copy' option
+        # during motion correction, although be careful about the components near
+        # the boundaries
+        
+        # memory map the file in order 'C'
+        fname_new = cm.save_memmap_join(mc.mmap_file, base_name='memmap_' + fnames.split('/')[-1].split('.')[0],
+                                        add_to_mov=border_to_0, dview=dview)  # exclude border
+    else: 
+        mmap_list = [file for file in os.listdir(file_dir) if ('memmap_' + fnames.split('/')[-1].split('.')[0]) in file]
+        fname_new = os.path.join(file_dir, mmap_list[0])
+        print('reuse previously saved memory mapping file:', fname_new)
     
-    # memory map the file in order 'C'
-    fname_new = cm.save_memmap_join(mc.mmap_file, base_name='memmap_',
-                                    add_to_mov=border_to_0, dview=dview)  # exclude border
-
 # %% SEGMENTATION
     # create summary images
     img = mean_image(mc.mmap_file[0], window = 1000, dview=dview)
     img = (img-np.mean(img))/np.std(img)
     
-    gaussian_blur = False        # Use gaussian blur when the quality of corr image(Cn) is bad
+    gaussian_blur = False        # Use gaussian blur when there is too much noise in the video
     Cn = local_correlations_movie_offline(mc.mmap_file[0], fr=fr, window=fr*4, 
                                           stride=fr*4, winSize_baseline=fr, 
                                           remove_baseline=True, gaussian_blur=gaussian_blur,
                                           dview=dview).max(axis=0)
     img_corr = (Cn-np.mean(Cn))/np.std(Cn)
-    summary_image = np.stack([img, img, img_corr], axis=2).astype(np.float32) 
+    summary_images = np.stack([img, img, img_corr], axis=0).astype(np.float32)
+    # ! save summary image, it is used in GUI
+    cm.movie(summary_images).save(fnames[:-5] + '_summary_images.tif')
     
     #%% three methods for segmentation
     methods_list = ['manual_annotation',        # manual annotation needs user to prepare annotated datasets same format as demo ROIs 
-                    'quick_annotation',         # quick annotation annotates data with simple interface in python
-                    'maskrcnn' ]                # maskrcnn is a convolutional network trained for finding neurons using summary images
-    method = methods_list[0]
+                    'gui_annotation',          # use gui to manual annotate neurons (it can perform basic functions but still under developing)
```

Comment:
to manually annotate ...
but this is still under development

Reply:
done